### PR TITLE
Add bulk create and batch release resource that uses it

### PR DESCRIPTION
**Pull request by kristina-thatch**
😰 <- me while just wingin' it on extending this client

Note mostly to self: to test, do something like this in the gemfile `gem "unit-ruby", path: "../../thatch-health/unit-ruby"`

```
Unit::BatchRelease.bulk_create([
  {
    amount: 123,
    description: "rel1",
    sender_name: "K S",
    sender_address: {
      street: "5230 Newell Rd",
      city: "Palo Alto",
      state: "CA",
      postal_code: "94303",
      country: "US",
    },
    sender_account_number: "asdf1",
    relationships: {
      batch_account: {
        data: {
          type: "batchAccount",
          id: "869264",
        }
      },
      receiver: {
        data: {
          type: "depositAccount",
          id: "804348",
        }
      }
    }
  },
  {
    amount: 345,
    description: "rel2",
    sender_name: "K S",
    sender_address: {
      street: "5230 Newell Rd",
      city: "Palo Alto",
      state: "CA",
      postal_code: "94303",
      country: "US",
    },
    sender_account_number: "asdf1",
    relationships: {
      batch_account: {
        data: {
          type: "batchAccount",
          id: "869264",
        }
      },
      receiver: {
        data: {
          type: "depositAccount",
          id: "701984",
        }
      }
    }
  },
])
```

```
[#<Unit::BatchRelease:0x000000010c2b4fc8
  @amount=123,
  @description="rel1",
  @dirty_attributes=[],
  @id="2053564",
  @idempotency_key=nil,
  @raw_data=
   {:type=>"batchRelease",
    :id=>"2053564",
    :attributes=>
     {:sender_name=>"K S",
      :sender_address=>
       {:street=>"5230 Newell Rd",
        :city=>"Palo Alto",
        :state=>"CA",
        :postal_code=>"94303",
        :country=>"US"},
      :sender_account_number=>"asdf1",
      :amount=>123,
      :description=>"rel1"},
    :relationships=>
     {:batch_account=>{:data=>{:type=>"batchAccount", :id=>"869264"}},
      :receiver=>{:data=>{:type=>"account", :id=>"804348"}}}},
  @relationships=
   {:batch_account=>{:data=>{:type=>"batchAccount", :id=>"869264"}},
    :receiver=>{:data=>{:type=>"account", :id=>"804348"}}},
  @sender_account_number="asdf1",
  @sender_address=
   #<Unit::Types::Address:0x000000010c2b41b8
    @city="Palo Alto",
    @country="US",
    @postal_code="94303",
    @state="CA",
    @street="5230 Newell Rd",
    @street2=nil>,
  @sender_name="K S",
  @tags=#<Unit::Types::Hash:0x000000010c2b4460 @items={}>,
  @type="batchRelease">,
 #<Unit::BatchRelease:0x000000010c2afaa0
  @amount=345,
  @description="rel2",
  @dirty_attributes=[],
  @id="2053565",
  @idempotency_key=nil,
  @raw_data=
   {:type=>"batchRelease",
    :id=>"2053565",
    :attributes=>
     {:sender_name=>"K S",
      :sender_address=>
       {:street=>"5230 Newell Rd",
        :city=>"Palo Alto",
        :state=>"CA",
        :postal_code=>"94303",
        :country=>"US"},
      :sender_account_number=>"asdf1",
      :amount=>345,
      :description=>"rel2"},
    :relationships=>
     {:batch_account=>{:data=>{:type=>"batchAccount", :id=>"869264"}},
      :receiver=>{:data=>{:type=>"account", :id=>"701984"}}}},
  @relationships=
   {:batch_account=>{:data=>{:type=>"batchAccount", :id=>"869264"}},
    :receiver=>{:data=>{:type=>"account", :id=>"701984"}}},
  @sender_account_number="asdf1",
  @sender_address=
   #<Unit::Types::Address:0x000000010c2aeb28
    @city="Palo Alto",
    @country="US",
    @postal_code="94303",
    @state="CA",
    @street="5230 Newell Rd",
    @street2=nil>,
  @sender_name="K S",
  @tags=#<Unit::Types::Hash:0x000000010c2aedd0 @items={}>,
  @type="batchRelease">]
  ```

### CODE REVIEW

File: lib/unit-ruby/util/resource_operations.rb [hunk 1]
```diff
@@ -56,6 +56,36 @@ def create(attributes)
...
+          end
+
+          created_resource_list = connection.post(resources_path, { data: data_list })
+
+          created_resource_list.map do |created_resource|
+            build_resource_from_json_api(created_resource)
+          end
+        end
+      end
+    end
```
Comment (kristina-thatch) [inline lib/unit-ruby/util/resource_operations.rb:87]: this is a copy-pasta'd version of `create` above but now handling a list of inputs + outputs

File: lib/unit-ruby/util/api_resource.rb [hunk 2]
```diff
@@ -83,8 +83,9 @@ def resource_type
 
     # Creates an association to a related resource
     # This will create a helper method to traverse into a resource's related resource(s)
-    def self.belongs_to(resource_name, class_name: nil)
+    def self.belongs_to(resource_name, resource_type: nil, class_name: nil)
       class_name ||= resource_name.to_s.camelize
+      resource_type ||= resource_name.to_s.camelize(:lower)
```
Comment (kristina-thatch) [inline lib/unit-ruby/util/api_resource.rb:88]: this is kinda just future proofing -- all `belongs_to` associations right now are either a single lowercase resource_name, or have an explicit resource_type defined, but in theory this should let you do something like `belongs_to :application_form` and `resource_type` will be `applicationForm` without being explicitly defined

File: lib/unit-ruby/util/api_resource.rb [hunk 3]
```diff
@@ -96,7 +97,7 @@ def self.belongs_to(resource_name, class_name: nil)
 
       define_method("#{resource_name}=") do |resource|
         relationships[resource_name] = {
-          data: { type: resource_name, id: resource.id }
+          data: { type: resource_type || resource_name, id: resource.id }
```
Comment (bdewater) [inline lib/unit-ruby/util/api_resource.rb]: Shouldn't this be just `resource_type` since it's always a non-nil value from line 88?

Comment (kristina-thatch) [reply lib/unit-ruby/util/api_resource.rb]: yess 100%, good catch 😬

**Pull request merged**